### ISSUE TITLE
[BUG] On CPU data loaders should only use pandas (not cudf)

### ISSUE THREAD

**Issue by karlhigley** [NVIDIA-Merlin/models#179]
**Describe the bug**
When setting the `device="cpu"` flag on `tf_dataloader.Dataset`, loading a dataset causes errors in `_create_tensors` related to type mismatches between pandas and cudf:

```
tests/unit/inference/test_poc.py:254: in create_retrieval_model
    history = model.fit(train_dataset_tf, callbacks=[tensorboard_callback], epochs=1)
/usr/local/lib/python3.8/dist-packages/keras/utils/traceback_utils.py:67: in error_handler
    raise e.with_traceback(filtered_tb) from None
/usr/local/lib/python3.8/dist-packages/merlin_models/tf/dataset.py:287: in __getitem__
    return DataLoader.__next__(self)
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:352: in __next__
    return self._get_next_batch()
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:385: in _get_next_batch
    self._fetch_chunk()
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:364: in _fetch_chunk
    raise chunks
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:158: in load_chunks
    self.chunk_logic(itr)
/usr/local/lib/python3.8/dist-packages/nvtx/nvtx.py:101: in inner
    result = func(*args, **kwargs)
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:138: in chunk_logic
    chunks = self.dataloader.make_tensors(chunks, self.dataloader._use_nnz)
/usr/local/lib/python3.8/dist-packages/nvtx/nvtx.py:101: in inner
    result = func(*args, **kwargs)
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:414: in make_tensors
    chunks = self._create_tensors(gdf)
/usr/local/lib/python3.8/dist-packages/nvtx/nvtx.py:101: in inner
    result = func(*args, **kwargs)
/usr/local/lib/python3.8/dist-packages/merlin_models/loader/backend.py:592: in _create_tensors
    offsets[column_name] = col_offsets.reset_index(drop=True)
/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:3607: in __setitem__
    self._set_item(key, value)
/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:3779: in _set_item
    value = self._sanitize_column(value)
/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:4497: in _sanitize_column
    self._ensure_valid_index(value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Empty DataFrame
Columns: []
Index: []
value = 0                0
1                3
2                4
3                5
4                7
            ...   
541868    11507633
541869    11507634
541870    11507663
541871    11507684
541872    11507693
Length: 541873, dtype: int64

    def _ensure_valid_index(self, value) -> None:
        """
        Ensure that if we don't have an index, that we can create one from the
        passed value.
        """
        # GH5632, make sure that we are a Series convertible
        if not len(self.index) and is_list_like(value) and len(value):
            if not isinstance(value, DataFrame):
                try:
                    value = Series(value)
                except (ValueError, NotImplementedError, TypeError) as err:
>                   raise ValueError(
                        "Cannot set a frame with no defined index "
                        "and a value that cannot be converted to a Series"
                    ) from err
E                   ValueError: Cannot set a frame with no defined index and a value that cannot be converted to a Series

/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py:3843: ValueError
```

**Steps/Code to reproduce bug**
Load a dataset on CPU and set the `device="cpu"` option:
```
    train_dataset_tf = tf_dataloader.Dataset(
        os.path.join(INPUT_DATA_DIR, "retrieval_training.parquet"),  
        batch_size=16,
        label_names=["target_item"],
        cat_names=[] + ["search_terms", "movie_ids", "genres"],
        cont_names=["movie_id_count"],
        engine="parquet",
        shuffle=True,
        buffer_size=0.25,
        parts_per_chunk=1,
        device="cpu",
    )
```

**Expected behavior**
When `device="cpu"` is set, dataset/data loader should create tensors with pandas instead of cudf.

**Additional context**
A potential fix would be to pass the device parameter from `loader.backend.DataLoader._create_tensors()` down through `dispatch._pull_apart_list` into `dispatch._make_series`, so that method knows which framework to use.

**Comment by karlhigley** [comment]
Related NVT PR: https://github.com/NVIDIA-Merlin/NVTabular/pull/1429

**Comment by karlhigley** [comment]
Fixed by #222
